### ISSUE TITLE
Hardcoded `gpg` command in "keybase/cURL/bash" scripts is problematic

### ISSUE THREAD

**Issue by crosser** [keybase/keybase-issues#2452]
I am not sure how much of a "bug" it is, but it did cause me some head scratching, so maybe it should be at least somewhat documented.
I keep my subkeys on Yubikey NEO, which is not well supported by gpg 1.4. So I have both `gpg` and `gpg2` installed, and `gpg2` *must* be used to operate my private keys. When I first tried to copy-paste the "keybase/cURL/bash" scripts, I got some errors to the effect of "private key not found", because gpg 1.4 does not have access to the keys on the card. I had to save the commands into a file and change the command name. After that the script worked.
I do not see a "proper" way to fix it, but maybe a hint on the web page that produces the script could help.

**Comment by crosser** [comment]
Apparently, command-line `keybase` suffers from the same problem:

```
crosser@pccross:~$ keybase pgp sign -m "this is a test signed message"
▶ ERROR No secret key available
```

**Comment by maxtaco** [comment]
I disagree, it's likely because you haven't `keybase pgp select`ed or `keybase pgp import`ed

**Comment by crosser** [comment]
```
crosser@pccross:~$ keybase pgp list
Keybase Key ID:  010153f1ba810d37e5294697ef0adf5dda5d29579406ff05ab9e5659ea7a04aa91c40a
PGP Fingerprint: D53E 85EF 3ED4 36C7 78DD D5AE A904 1071 00FC C595
PGP Identities:
   Eugene Crosser (Evgeny Cherkashin) <crosser@average.org>
   Evgeny Cherkashin (Software Engineer at IBM) <Eugene.Crosser@ru.ibm.com>
```

**Comment by maxtaco** [comment]
Ah ok!  We don't support shelling out to GPG for signing right now.  I'd say continue to do that with GPG.  It's an advanced feature so we figured you wouldn't mind :)

**Comment by crosser** [comment]
OK. :)
The original request still stands.

**Comment by pafcu** [comment]
It seems like it should be fairly simply to check if the `gpg` command failed, and if it did, try with `gpg2`instead. I also ran into the same problem: simply changing the call to `gpg2` from `gpg` fixed it.

**Comment by mazerty** [comment]
i encountered a similar problem today when i added my public key
the "version" header in it was "GnuPG v2" yet the curl command asked for gpg and failed
maybe you can check the version and adapt the curl command accordingly ?
